File: solutions/lua/nth-prime/3/nth-prime.lua
return function(number)
	if number <= 0 then
		error("")
	end

	local ps = {}
	local i = 2
	while #ps < number do
		local is_prime = true
		for j = 1, #ps do
			local p = ps[j]
			if p*p > i then
				break
			end
			if i % p == 0 then
				is_prime = false
				break
			end
		end
		if is_prime then
			table.insert(ps, i)
		end
		i = i + 1
	end
	return ps[#ps]
end

File: solutions/ocaml/nth-prime/1/nth_prime.ml
let nth_prime number =
  if number == 0 then
    Error "there is no zeroth prime"
  else
  let rec aux ps i =
    if number <= List.length ps then
      List.hd ps
    else
      let is_prime = List.for_all (fun p -> (i mod p) <> 0) ps in
      if is_prime then
        aux (i :: ps) (i + 1)
      else
        aux ps (i + 1)
  in Ok (aux [] 2)
